include/task.h

```diff
@@ -267,6 +278,8 @@ namespace detail {
 		reduction_set m_reductions;
 		std::string m_debug_name;
 		detail::epoch_action m_epoch_action;
+		// TODO I believe that `struct task` should not store command_group_launchers, fence_promise or other state that is related to execution instead of
+		// abstract DAG building. For user-initialized buffers we already notify the runtime -> executor of this state directly. Maybe also do that for these.
```

Comment:
Having them in the task graph means we also need to copy them into kernel instructions, because the IDAG does not reference tasks (by pointer or id). This is another entry in the long list of refactoring goals for the task manager and task graph.